hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/DirectoryDeletingService.java

```diff
@@ -125,73 +134,95 @@ public int getPriority() {
+                    pendingDeletedDirInfo.getValue(), remainNum);
+            remainNum = remainNum - subFiles.size();
 
             if (LOG.isDebugEnabled()) {
-              for (OmKeyInfo fileInfo : purgeDeletedFiles) {
-                LOG.debug("deleted sub file name: {}", fileInfo.getKeyName());
+              for (OmKeyInfo fileInfo : subFiles) {
+                LOG.debug("Moved sub file name: {}", fileInfo.getKeyName());
               }
             }
 
-            // step-3: Since there is a boundary condition of 'numEntries' in
-            // each batch, check whether the sub paths count reached batch size
-            // limit. If count reached limit then there can be some more child
-            // paths to be visited and will keep the parent deleted directory
-            // for one more pass.
-            final Optional<String> purgeDeletedDir = count > 0 ?
-                    Optional.of(pendingDeletedDirInfo.getKey()) :
-                    Optional.empty();
-
-            if (isRatisEnabled()) {
-              submitPurgePaths(volumeId, bucketId, purgeDeletedDir,
-                      purgeDeletedFiles, deletedSubDirList);
+          // step-3: Since there is a boundary condition of 'numEntries' in
+          // each batch, check whether the sub paths count reached batch size
+          // limit. If count reached limit then there can be some more child
+          // paths to be visited and will keep the parent deleted directory
+          // for one more pass.
+            final Optional<String> purgeDeletedDir = remainNum > 0 ?
+                Optional.of(pendingDeletedDirInfo.getKey()) :
+                Optional.empty();
+
+            PurgePathRequest request = wrapPurgeRequest(volumeId, bucketId,
+                purgeDeletedDir, subFiles, subDirs);
+            purgePathRequestList.add(request);
+
+            // Count up the purgeDeletedDir, subDirs and subFiles
+            if (purgeDeletedDir.isPresent()) {
+              dirNum++;
             }
-            // TODO: need to handle delete with non-ratis
+            subDirNum += subDirs.size();
+            subFileNum += subFiles.size();
+          }
 
-            deletedDirsCount.incrementAndGet();
-            deletedFilesCount.addAndGet(purgeDeletedFiles.size());
-            if (LOG.isDebugEnabled()) {
-              LOG.debug("Number of dirs deleted: {}, Number of files moved:" +
-                      " {} to DeletedTable, elapsed time: {}ms",
-                  deletedDirsCount, deletedFilesCount,
-                  Time.monotonicNow() - startTime);
-            }
+          // TODO: need to handle delete with non-ratis
+          if (isRatisEnabled()) {
+            submitPurgePaths(purgePathRequestList);
           }
+
+          deletedDirsCount.addAndGet(dirNum);
+          movedDirsCount.addAndGet(subDirNum);
+          movedFilesCount.addAndGet(subFileNum);
+          LOG.info("Number of dirs deleted: {}, Number of sub-files moved:" +
```

Comment:
NIT: maybe debug here is better than info

Reply:
Make sense! I have updated.

Reply:
Hi, Jackson! After some thought, and according to practice, such conclusion info should be logged stead of debugged, such msg will not fill too much, but useful!